hw/arm/xlnx-versal-virt.c

```diff
@@ -276,15 +276,22 @@ static void fdt_add_canfd_nodes(VersalVirt *s)
     }
 }
 
-static void fdt_add_fixed_link_nodes(VersalVirt *s, char *gemname,
+static void fdt_add_generic_phy_node(VersalVirt *s, char *gemname,
                                      uint32_t phandle)
 {
-    char *name = g_strdup_printf("%s/fixed-link", gemname);
+    uint64_t addr = MM_GEM_PHY_OFF;
+    char *name = g_strdup_printf("%s/ethernet-phy@%" PRIx64, gemname, addr);
+    const char compat_phy[] = "genericPhy";
 
     qemu_fdt_add_subnode(s->fdt, name);
     qemu_fdt_setprop_cell(s->fdt, name, "phandle", phandle);
-    qemu_fdt_setprop(s->fdt, name, "full-duplex", NULL, 0);
-    qemu_fdt_setprop_cell(s->fdt, name, "speed", 1000);
+    qemu_fdt_setprop_cell(s->fdt, name, "fifo-depth", 0x01);
```

Comment:
Should there be #defines or other constant definitions for 0x01, 0a0a, 0x08, 0x03 (the various constants on this line through line 292)?

Reply:
actually, it looks like these properties are not needed when initializing the genericPhy driver. I had borrowed them from the zynqmp device tree, but on the real hardware that phy binds to the tiTdPhy driver; on the qemu platform the phy is not fully implemented and vxworks falls back to the genericPhy driver.

i'll go ahead and remove these other properties since they have no meaning to the emulated phy.